test/framework/src/main/java/org/elasticsearch/plugins/MockPluginsService.java

```diff
@@ -76,4 +80,43 @@ public PluginsAndModules info() {
             List.of()
         );
     }
+
+    @Override
+    public <T> List<? extends T> loadServiceProviders(Class<T> service) {
```

Comment:
Can this be moved out to a separate PR, with a unit test?

Reply:
Sounds good, I made a new PR here https://github.com/elastic/elasticsearch/pull/88690. Once that's merged I'll rebase and remove this change.